File: lib/screens/quiz_flow_selection_screen.dart
import 'package:flutter/material.dart';
import '../components/quizora_components.dart';
import '../data/database_helper.dart';
import '../data/cbse_content_provider.dart';
import 'study_main_screen.dart';

enum SelectionStep { clazz, subject, chapter }

class QuizFlowSelectionScreen extends StatefulWidget {
  final bool initialIsNotesFlow;

  const QuizFlowSelectionScreen({Key? key, required this.initialIsNotesFlow}) : super(key: key);

  @override
  _QuizFlowSelectionScreenState createState() => _QuizFlowSelectionScreenState();
}

class _QuizFlowSelectionScreenState extends State<QuizFlowSelectionScreen> {
  SelectionStep _currentStep = SelectionStep.clazz;
  String _selectedClass = 'Class 10';
  String _selectedSubject = 'Science';
  String _selectedLanguage = 'English';
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _loadUserPreferences();
  }

  Future<void> _loadUserPreferences() async {
    final progress = await DatabaseHelper.instance.getProgress();
    setState(() {
      _selectedClass = progress.selectedClass;
      _selectedLanguage = progress.selectedLanguage.isEmpty ? 'English' : progress.selectedLanguage;
      _isLoading = false;
    });
  }

  @override
  Widget build(BuildContext context) {
    if (_isLoading) {
      return const Center(child: CircularProgressIndicator(color: neonBlue));
    }

    // Dynamic subjects
    final isSeniorClass = _selectedClass == 'Class 11' || _selectedClass == 'Class 12';
    final List<String> subjectsList = isSeniorClass
        ? ['Physics', 'Chemistry', 'Biology', 'Mathematics', 'English', 'Hindi']
        : ['Science', 'Mathematics', 'English', 'Social Science', 'Hindi'];

    // Dynamic chapters
    final List<String> chaptersList = CBSEContentProvider.getChapters(_selectedClass, _selectedSubject, _selectedLanguage);

    return Scaffold(
      backgroundColor: Colors.transparent,
      body: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 20.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 16),
            if (widget.initialIsNotesFlow && _currentStep == SelectionStep.clazz) ...[
              const Text(
                'SELECT BOOK NOTES',
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 18,
                  fontWeight: FontWeight.black,
                  letterSpacing: 0.5,
                ),
              ),
              const SizedBox(height: 4),
              const Text(
                'Notes are paired along with quizzes. Select any chapter code below to access instant notes summary.',
                style: TextStyle(
                  color: textSecondary,
                  fontSize: 12,
                ),
              ),
              const SizedBox(height: 16),
            ],

            // Step Header and back navigation
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Row(
                  children: [
                    if (_currentStep != SelectionStep.clazz)
                      IconButton(
                        icon: const Icon(Icons.arrow_back, color: neonBlue),
                        onPressed: () {
                          setState(() {
                            _currentStep = _currentStep == SelectionStep.chapter
                                ? SelectionStep.subject
                                : SelectionStep.clazz;
                          });
                        },
                      ),
                    Text(
                      _getStepTitle(),
                      style: const TextStyle(
                        color: Colors.white,
                        fontSize: 18,
                        fontWeight: FontWeight.black,
                      ),
                    ),
                  ],
                ),
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 10.0, vertical: 4.0),
                  decoration: BoxDecoration(
                    color: neonPurple.withOpacity(0.15),
                    borderRadius: BorderRadius.circular(8.0),
                  ),
                  child: Text(
                    '${_getStepIndex()}/3',
                    style: const TextStyle(
                      color: neonPurple,
                      fontSize: 13,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 8),

            // Pills summary of selection
            if (_currentStep != SelectionStep.clazz)
              Row(
                children: [
                  _SelectionProgressPill(label: 'Class', value: _selectedClass),
                  const SizedBox(width: 8),
                  if (_currentStep == SelectionStep.chapter)
                    _SelectionProgressPill(label: 'Subject', value: _selectedSubject),
                ],
              ),
            const SizedBox(height: 16),

            // Step wizard list or grid
            Expanded(
              child: AnimatedSwitcher(
                duration: const Duration(milliseconds: 180),
                child: _buildStepContent(subjectsList, chaptersList),
              ),
            ),
          ],
        ),
      ),
    );
  }

  String _getStepTitle() {
    switch (_currentStep) {
      case SelectionStep.clazz:
        return 'STEP 1: SELECT CLASS';
      case SelectionStep.subject:
        return 'STEP 2: SELECT SUBJECT';
      case SelectionStep.chapter:
        return 'STEP 3: SELECT CHAPTER';
    }
  }

  int _getStepIndex() {
    switch (_currentStep) {
      case SelectionStep.clazz:
        return 1;
      case SelectionStep.subject:
        return 2;
      case SelectionStep.chapter:
        return 3;
    }
  }

  Widget _buildStepContent(List<String> subjectsList, List<String> chaptersList) {
    switch (_currentStep) {
      case SelectionStep.clazz:
        final classes = CBSEContentProvider.getClasses();
        return GridView.builder(
          key: const ValueKey('class_grid'),
          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
            crossAxisCount: 2,
            crossAxisSpacing: 14,
            mainAxisSpacing: 14,
            childAspectRatio: 1.8,
          ),
          itemCount: classes.length,
          itemBuilder: (context, index) {
            final className = classes[index];
            final isSelected = _selectedClass == className;
            return _ClassSelectionCard(
              className: className,
              isSelected: isSelected,
              onClick: () async {
                final progress = await DatabaseHelper.instance.getProgress();
                await DatabaseHelper.instance.saveProgress(
                  progress.copyWith(selectedClass: className),
                );
                setState(() {
                  _selectedClass = className;
                  _currentStep = SelectionStep.subject;
                });
              },
            );
          },
        );

      case SelectionStep.subject:
        return GridView.builder(
          key: const ValueKey('subject_grid'),
          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
            crossAxisCount: 2,
            crossAxisSpacing: 14,
            mainAxisSpacing: 14,
            childAspectRatio: 1.8,
          ),
          itemCount: subjectsList.length,
          itemBuilder: (context, index) {
            final subjectName = subjectsList[index];
            final isSelected = _selectedSubject == subjectName;
            return _SubjectSelectionCard(
              subjectName: subjectName,
              isSelected: isSelected,
              onClick: () {
                setState(() {
                  _selectedSubject = subjectName;
                  _currentStep = SelectionStep.chapter;
                });
              },
            );
          },
        );

      case SelectionStep.chapter:
        if (chaptersList.isEmpty) {
          return const Center(
            child: Text(
              'No chapters available for this configuration.',
              style: TextStyle(color: textSecondary),
            ),
          );
        }
        return ListView.builder(
          key: const ValueKey('chapter_list'),
          itemCount: chaptersList.length,
          itemBuilder: (context, index) {
            final chapterName = chaptersList[index];
            return _ChapterSelectionCard(
              chapterName: chapterName,
              onClick: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(
                    builder: (context) => StudyMainScreen(
                      subject: _selectedSubject,
                      chapter: chapterName,
                      isDailyChallenge: false,
                      initialTabIndex: widget.initialIsNotesFlow ? 0 : 1,
                    ),
                  ),
                );
              },
            );
          },
        );
    }
  }
}

class _SelectionProgressPill extends StatelessWidget {
  final String label;
  final String value;

  const _SelectionProgressPill({
    Key? key,
    required this.label,
    required this.value,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10.0, vertical: 6.0),
      decoration: BoxDecoration(
        color: glassBg,
        borderRadius: BorderRadius.circular(8.0),
      ),
      child: Text(
        '$label: $value',
        style: const TextStyle(
          color: textSecondary,
          fontSize: 11,
          fontWeight: FontWeight.bold,
        ),
      ),
    );
  }
}

class _ClassSelectionCard extends StatelessWidget {
  final String className;
  final bool isSelected;
  final VoidCallback onClick;

  const _ClassSelectionCard({
    Key? key,
    required this.className,
    required this.isSelected,
    required this.onClick,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final glowColor = isSelected ? neonBlue : textMuted;
    return Container(
      decoration: BoxDecoration(
        color: isSelected ? glowColor.withOpacity(0.2) : glassBg,
        borderRadius: BorderRadius.circular(16.0),
        border: Border.all(
          color: isSelected ? glowColor : glowColor.withOpacity(0.2),
          width: 1.0,
        ),
      ),
      child: Material(
        color: Colors.transparent,
        child: InkWell(
          borderRadius: BorderRadius.circular(16.0),
          onTap: onClick,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Text(
                className,
                style: const TextStyle(
                  color: Colors.white,
                  fontSize: 18,
                  fontWeight: FontWeight.black,
                ),
              ),
              Text(
                'CBSE',
                style: TextStyle(
                  color: isSelected ? neonBlue : textSecondary,
                  fontSize: 10,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

class _SubjectSelectionCard extends StatelessWidget {
  final String subjectName;
  final bool isSelected;
  final VoidCallback onClick;

  const _SubjectSelectionCard({
    Key? key,
    required this.subjectName,
    required this.isSelected,
    required this.onClick,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final glowColor = isSelected ? neonPurple : textMuted;
    return Container(
      decoration: BoxDecoration(
        color: isSelected ? glowColor.withOpacity(0.2) : glassBg,
        borderRadius: BorderRadius.circular(16.0),
        border: Border.all(
          color: isSelected ? glowColor : glowColor.withOpacity(0.2),
          width: 1.0,
        ),
      ),
      child: Material(
        color: Colors.transparent,
        child: InkWell(
          borderRadius: BorderRadius.circular(16.0),
          onTap: onClick,
          child: Container(
            alignment: Alignment.center,
            padding: const EdgeInsets.all(8.0),
            child: Text(
              subjectName,
              style: const TextStyle(
                color: Colors.white,
                fontSize: 16,
                fontWeight: FontWeight.black,
              ),
              textAlign: TextAlign.center,
            ),
          ),
        ),
      ),
    );
  }
}

class _ChapterSelectionCard extends StatelessWidget {
  final String chapterName;
  final VoidCallback onClick;

  const _ChapterSelectionCard({
    Key? key,
    required this.chapterName,
    required this.onClick,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: 12.0),
      decoration: BoxDecoration(
        color: glassBg,
        borderRadius: BorderRadius.circular(16.0),
        border: Border.all(
          color: neonBlue.withOpacity(0.4),
          width: 1.0,
        ),
      ),
      child: Material(
        color: Colors.transparent,
        child: InkWell(
          borderRadius: BorderRadius.circular(16.0),
          onTap: onClick,
          child: Padding(
            padding: const EdgeInsets.all(20.0),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        chapterName,
                        style: const TextStyle(
                          color: Colors.white,
                          fontSize: 15,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                      const SizedBox(height: 4),
                      const Text(
                        'Syllabus Aligned MCQ Quiz & Notes',
                        style: TextStyle(
                          color: textSecondary,
                          fontSize: 11,
                        ),
                      ),
                    ],
                  ),
                ),
                const Icon(
                  Icons.chevron_right,
                  color: neonBlue,
                  size: 24,
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/screens/QuizFlowSelectionScreen.kt
package com.example.ui.screens

import androidx.compose.animation.*
import androidx.compose.animation.core.tween
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.GlassCard
import com.example.ui.components.NeonButton
import com.example.ui.theme.*
import com.example.viewmodel.QuizoraViewModel
import com.example.data.CBSEContentProvider

enum class SelectionStep {
    CLASS, SUBJECT, CHAPTER
}

@Composable
fun QuizFlowSelectionScreen(
    viewModel: QuizoraViewModel,
    onNavigateToStudy: () -> Unit
) {
    var currentStep by remember { mutableStateOf(SelectionStep.CLASS) }
    
    val selectedClass by viewModel.currentClass.collectAsState()
    val selectedSubject by viewModel.currentSubject.collectAsState()
    val selectedChapter by viewModel.currentChapter.collectAsState()
    val progress by viewModel.userProgress.collectAsState()
    val lang = progress.selectedLanguage.ifEmpty { "English" }

    val classesList = listOf("Class 6", "Class 7", "Class 8", "Class 9", "Class 10", "Class 11", "Class 12")

    // Dynamic subject lists
    val isSeniorClass = selectedClass == "Class 11" || selectedClass == "Class 12"
    val subjectsList = if (isSeniorClass) {
        listOf("Physics", "Chemistry", "Biology", "Mathematics", "English", "Hindi")
    } else {
        listOf("Science", "Mathematics", "English", "Social Science", "Hindi")
    }

    // Dynamic chapter lists (pulling from dynamic CBSEContentProvider)
    val chaptersList = remember(selectedClass, selectedSubject, lang) {
        CBSEContentProvider.getChapters(selectedClass, selectedSubject, lang)
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Spacer(modifier = Modifier.height(12.dp))

        // --- Wizard Banner & Back Navigation ---
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                if (currentStep != SelectionStep.CLASS) {
                    IconButton(
                        onClick = {
                            currentStep = when (currentStep) {
                                SelectionStep.SUBJECT -> SelectionStep.CLASS
                                SelectionStep.CHAPTER -> SelectionStep.SUBJECT
                                else -> SelectionStep.CLASS
                            }
                        },
                        modifier = Modifier.testTag("step_back_button")
                    ) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Step Back", tint = NeonBlue)
                    }
                }
                Text(
                    text = when (currentStep) {
                        SelectionStep.CLASS -> "STEP 1: SELECT CLASS"
                        SelectionStep.SUBJECT -> "STEP 2: SELECT SUBJECT"
                        SelectionStep.CHAPTER -> "STEP 3: SELECT CHAPTER"
                    },
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Black,
                    color = Color.White
                )
            }

            // Step Progress Chips
            Text(
                text = "${currentStep.ordinal + 1}/3",
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                color = NeonPurple,
                modifier = Modifier
                    .background(NeonPurple.copy(alpha = 0.15f), shape = RoundedCornerShape(8.dp))
                    .padding(horizontal = 10.dp, vertical = 4.dp)
            )
        }

        // Selected Status Summary
        if (currentStep != SelectionStep.CLASS) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                SelectionProgressPill(label = "Class", value = selectedClass)
                if (currentStep == SelectionStep.CHAPTER) {
                    SelectionProgressPill(label = "Subject", value = selectedSubject)
                }
            }
        }

        // Content Area with smooth transitions
        Box(modifier = Modifier.weight(1f)) {
            AnimatedContent(
                targetState = currentStep,
                transitionSpec = {
                    fadeIn(animationSpec = tween(180)) togetherWith fadeOut(animationSpec = tween(180))
                },
                label = "step_wizard_content"
            ) { targetStep ->
                when (targetStep) {
                    SelectionStep.CLASS -> {
                        LazyVerticalGrid(
                            columns = GridCells.Fixed(2),
                            verticalArrangement = Arrangement.spacedBy(14.dp),
                            horizontalArrangement = Arrangement.spacedBy(14.dp),
                            contentPadding = PaddingValues(bottom = 16.dp)
                        ) {
                            items(classesList) { className ->
                                ClassSelectionCard(
                                    className = className,
                                    isSelected = selectedClass == className,
                                    onClick = {
                                        viewModel.selectClass(className)
                                        currentStep = SelectionStep.SUBJECT
                                    }
                                )
                            }
                        }
                    }
                    SelectionStep.SUBJECT -> {
                        LazyVerticalGrid(
                            columns = GridCells.Fixed(2),
                            verticalArrangement = Arrangement.spacedBy(14.dp),
                            horizontalArrangement = Arrangement.spacedBy(14.dp),
                            contentPadding = PaddingValues(bottom = 16.dp)
                        ) {
                            items(subjectsList) { subjectName ->
                                SubjectSelectionCard(
                                    subjectName = subjectName,
                                    isSelected = selectedSubject == subjectName,
                                    onClick = {
                                        viewModel.selectSubject(subjectName)
                                        currentStep = SelectionStep.CHAPTER
                                    }
                                )
                            }
                        }
                    }
                    SelectionStep.CHAPTER -> {
                        LazyColumn(
                            verticalArrangement = Arrangement.spacedBy(12.dp),
                            contentPadding = PaddingValues(bottom = 16.dp)
                        ) {
                            items(chaptersList) { chapterName ->
                                ChapterSelectionCard(
                                    chapterName = chapterName,
                                    onClick = {
                                        viewModel.selectChapter(chapterName)
                                        onNavigateToStudy()
                                    }
                                )
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun SelectionProgressPill(label: String, value: String) {
    Box(
        modifier = Modifier
            .background(GlassBg, RoundedCornerShape(8.dp))
            .padding(horizontal = 10.dp, vertical = 6.dp)
    ) {
        Text(
            text = "$label: $value",
            fontSize = 11.sp,
            fontWeight = FontWeight.Bold,
            color = TextSecondary
        )
    }
}

@Composable
fun ClassSelectionCard(
    className: String,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    val glowColor = if (isSelected) NeonBlue else TextMuted
    val fillAlpha = if (isSelected) 0.35f else 0.15f
    
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .height(90.dp)
            .clickable(onClick = onClick)
            .testTag("class_card_$className"),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = if (isSelected) glowColor.copy(alpha = fillAlpha) else GlassBg),
        border = BorderStroke(
            1.dp,
            if (isSelected) glowColor else glowColor.copy(alpha = 0.2f)
        )
    ) {
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    text = className,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Black,
                    color = Color.White
                )
                Text(
                    text = "CBSE",
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (isSelected) NeonBlue else TextSecondary
                )
            }
        }
    }
}

@Composable
fun SubjectSelectionCard(
    subjectName: String,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    val glowColor = if (isSelected) NeonPurple else TextMuted
    
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .height(90.dp)
            .clickable(onClick = onClick)
            .testTag("subject_card_$subjectName"),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = if (isSelected) glowColor.copy(alpha = 0.3f) else GlassBg),
        border = BorderStroke(
            1.dp,
            if (isSelected) glowColor else glowColor.copy(alpha = 0.2f)
        )
    ) {
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = subjectName,
                fontSize = 16.sp,
                fontWeight = FontWeight.Black,
                color = Color.White,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(8.dp)
            )
        }
    }
}

@Composable
fun ChapterSelectionCard(
    chapterName: String,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .testTag("chapter_card_$chapterName"),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = GlassBg),
        border = BorderStroke(1.dp, NeonBlue.copy(alpha = 0.4f))
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(20.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = chapterName,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Syllabus Aligned MCQ Quiz & Notes",
                    fontSize = 11.sp,
                    color = TextSecondary
                )
            }
            Icon(
                imageVector = Icons.Default.ChevronRight,
                contentDescription = "Select Chapter",
                tint = NeonBlue,
                modifier = Modifier.size(24.dp)
            )
        }
    }
}

@Composable
fun NotesScreenWrapper(
    viewModel: QuizoraViewModel,
    onNavigateToStudy: () -> Unit
) {
    // Quick routing from the main selector direct to Notes tab inside Study screen
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(20.dp))
        Text(
            text = "SELECT BOOK NOTES",
            fontSize = 18.sp,
            fontWeight = FontWeight.Black,
            color = Color.White,
            modifier = Modifier.align(Alignment.Start)
        )
        Spacer(modifier = Modifier.height(2.dp))
        Text(
            text = "Notes are paired along with quizzes. Select any chapter code below to access instant notes summary.",
            fontSize = 12.sp,
            color = TextSecondary,
            modifier = Modifier.align(Alignment.Start)
        )
        Spacer(modifier = Modifier.height(16.dp))

        Box(modifier = Modifier.weight(1f)) {
            QuizFlowSelectionScreen(
                viewModel = viewModel,
                onNavigateToStudy = onNavigateToStudy
            )
        }
    }
}
